doc/arch/adr-002-form-representation.md

```diff
@@ -0,0 +1,95 @@
+# ADR 2: JSON for form representation and storage
+
+## Context
+
+Bugle forms needs a data model to support these uses:
+- Users create forms with an arbitrary number of questions.
+- Question forms can have a variety of field types—text, checkbox, radio buttons
+  and dropdown boxes. It should be easy to add more response types later.
+- Users provide an unknown number of responses for each question form. Each
+  response maps to the structure and requirements of its question form.
+- Sufficiently fast reads for retrieving form data and collating all answers in
+  the responses.
+
+The application currently uses PostgreSQL for storing user data.
+
+With that in mind, some options include:
+- Representing form questions and answers in JSON
+  - using PostgreSQL's `json` type for the column having the form data.
+  - using PostgreSQL's `text` type for the column having the form data.
+  - using PostgreSQL's `jsonb` type as the column having the form data.
+  - storing form data in a separate document store
+- Representing and storing form questions and answers in relations
+- Storing the form questions directly as HTML, and storing the answers in a relation
+
+## Decision
+
+We will use PostgreSQL's `jsonb` columns to store form questions and answers.
+The illustration below demonstrates how we will model the form data.
+
+![Data model](https://user-images.githubusercontent.com/24277692/156551390-af11e56a-f8d6-4d7f-a55b-b8d4e511104c.png)
+
+We still use relations for storing form and response IDs, along with their
+metadata. The form structure will be stored in a `jsonb` column.
```

Comment:
What is `metadata`? This hasn't been mentioned anywhere else so it'd be helpful to talk about this somewhere up top.

Reply:
Also something to consider: Is the shape of the `metadata` known beforehand? What is `meta` about the `metadata`? If it's just some data that changes shape depending on the question's type, are there other ways to model it?